code/modules/hydroponics/spreading/spreading.dm

```diff
@@ -0,0 +1,195 @@
+#define DEFAULT_SEED "glowshroom"
+#define CREEPER_GROWTH_DISTANCE 4
+
+/obj/effect/plantsegment
+	name = "space vines"
+	desc = "An extremely expansionistic species of vine."
+	icon = 'icons/effects/spacevines.dmi'
+	icon_state = "Light1"
+	anchored = 1
+	opacity = 0
+	density = 0
+	layer = 5
+	pass_flags = PASSTABLE | PASSGRILLE
+	//flags = PROXMOVE
+	//mouse_opacity = 2
+
+	var/health = 10
+	var/max_health = 100
+	var/list/neighbors = list()
+	var/turf/epicenter
+	var/datum/seed/seed
+	var/sampled = 0
+	var/spread_chance
+	var/spread_distance
+	var/mature_time
+	var/last_tick = 0
+	var/harvest = 0
+	var/age = 0
+	var/limited_growth = 0
+	var/plant_damage_noun = "Thorns"
+
+/obj/effect/plantsegment/creeper
+	limited_growth = 1
+/obj/effect/plantsegment/single
+	spread_chance = 0
+
+/obj/effect/plantsegment/Destroy()
+	if(plant_controller)
+		plant_controller.remove_plant(src)
+	for(var/obj/effect/plantsegment/neighbor in range(1,src))
+		plant_controller.add_plant(neighbor)
+	..()
+
+/obj/effect/plantsegment/New(var/newloc, var/datum/seed/newseed, var/turf/newepicenter, var/start_fully_mature = 0)
+	..()
+
+	if(!newepicenter)
+		epicenter = get_turf(src)
+	else
+		epicenter = newepicenter
+
+	if(!plant_controller)
+		sleep(250) // ugly hack, should mean roundstart plants are fine.
+	if(!plant_controller)
+		world << "<span class='danger'>Plant controller does not exist and [src] requires it. Aborting.</span>" //This probably shouldn't be a world <<
```

Comment:
I agree comment, if there's a lot of these things, and then the plant controller breaks...